patches/tModLoader/Terraria/ModLoader/Engine/InstallVerifier.cs

```diff
@@ -86,11 +86,12 @@ internal static class InstallVerifier
 				return false;
 			}
 
-			// Whether the steam_appid.txt file exists, indicating we'd have to check steam installation
-			if (File.Exists(SteamAppIDPath))
-				return CheckSteam();
+			// Check if we are within a steamapps subfolder OR are trying to force run as steam install; if not, return GoG.
+			if (!(Directory.GetCurrentDirectory().Contains("steamapps") || Program.LaunchParameters.ContainsKey("-steam"))) {
+				return CheckGoG();
```

Comment:
What if someone has a dev variant way outside steam? Instead this should check for whether or not steam_api.dll/unix variant (or maybe CSteamworks.dll) is present in the _vanilla_ game's folder (at `../` or `../Terraria/`). I've mentioned this in Discord.

Reply:
Had included the launch parameter for it, but had aimed to avoid rewriting large parts of InstallVerifier to facilitate. 
I've since enhanced it to check each of the steam_api variants & attempt to rule out GoG if Terraria folder is not nearby; although my certainty on unix steam_api location is limited, in particular for MacOS. 
~~Although feels more convoluted for this since can't just call the api to give steam Terraria install location. ~~